src/ThreeJsPanel.ts

```diff
@@ -605,6 +616,35 @@ export class ThreeJsPanel {
     return this.renderer.getContext().canvas.height;
   }
 
+  getCameraState(): CameraState {
+    return {
+      position: this.camera.position.toArray(),
+      up: this.camera.up.toArray(),
+      target: this.controls.target.toArray(),
+      orthoScale: this.camera.type === "OrthographicCamera" ? this.controls.scale : undefined,
+      fov: this.camera.type === "PerspectiveCamera" ? this.camera.fov : undefined,
+    };
+  }
+
+  setCameraState(state: Partial<CameraState>) {
+    const currentState = this.getCameraState();
+    // Fill in any missing properties with current state
+    const newState = { ...currentState, ...state };
+
+    this.camera.up.fromArray(newState.up).normalize();
+    this.camera.position.fromArray(newState.position);
+    this.controls.target.fromArray(newState.target);
+
+    // Update fields by camera type
+    if (this.camera.type === "OrthographicCamera") {
```

Comment:
prefer this.camera.isOrthographicCamera:  https://threejs.org/docs/#api/en/cameras/OrthographicCamera.isOrthographicCamera

Reply:
`isOrthographicCamera()` is only implemented on `OrthographicCamera` and not on `PerspectiveCamera`... so it's not actually a valid call unless I cast `this.camera`, and then the TypeScript conditional logic breaks down.

Reply:
We've already tackled that... you can call this func:  https://github.com/allen-cell-animated/volume-viewer/blob/98ff141423c4a94a8d8e8073e10fd0d5f2d4d708/src/types.ts#L89

Reply:
Thanks!